### PR TITLE
feat: anonymous frames

### PR DESCRIPTION
**Pull request by stephancill**
- Adds unauthenticated (anonymous) frame support

this approach loads the initial frame twice - frames.js limitation

todo
- [x] test with anonymous frame - https://x.com/stephancill/status/1807714793127612471

### CODE REVIEW

File: apps/server/hooks/use-anonymous-identity.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// import type { SignerStateInstance } from "@frames.js/render"
```
Comment (cryptods8) [inline apps/server/hooks/use-anonymous-identity.tsx]: extra?

**Review by cryptods8 — APPROVED** [review]
is there a plan to abstract it away and perhaps make it part of `useFrame`?

**Comment by stephancill** [comment]
@cryptods8 yeah i think that would be useful. the current abstraction doesn't play very well with clients that can handle multiple frame specs

File: apps/server/components/frame-render.tsx [hunk 2]
```diff
@@ -141,6 +157,32 @@ const useFrameDefaults = {
   specification: "farcaster" as const,
 };
 
+function useAnonymousFrame(props: FrameRenderWithIdentityProps) {
+  // TODO: find a way to do this without loading the frame twice
```
Comment (davidfurlong) [inline apps/server/components/frame-render.tsx:161]: using something like swr deduplicates requests automatically via caching - not sure if we have something similar in place we could use here

**Review by davidfurlong — APPROVED** [review]

File: apps/server/components/frame-render.tsx (hunk 2 above)
Comment (stephancill) [reply apps/server/components/frame-render.tsx:161]: it's technically not the same request. the one parses with open frames parser, the other parses with farcaster parser. if we move parsing to the client then we can use request caching

Comment (stephancill) [reply apps/server/components/frame-render.tsx:161]: merging for now. can fix this separately

**Pull request merged**